third_party/amd/lib/TritonAMDGPUToLLVM/DotOpToLLVM/MFMA.cpp

```diff
@@ -430,6 +460,284 @@ struct DotOpMFMAConversionHelper {
   }
 };
 
+struct ScaledDotOpMFMAConversionHelper : DotOpMFMAConversionHelper {
+
+  ScaledDotOpMFMAConversionHelper(AMDMfmaEncodingAttr mfmaLayout,
+                                  ConversionPatternRewriter &rewriter,
+                                  const LLVMTypeConverter *typeConverter,
+                                  Location loc)
+      : DotOpMFMAConversionHelper(mfmaLayout, rewriter, typeConverter, loc) {}
+
+  Value generateScaledMFMAOp(MfmaInsn &mfmaInsn, Value valA, Value valB,
+                             Value valC, Value valScaleA,
+                             Value valScaleB) const {
+    auto b = TritonLLVMOpBuilder(loc, rewriter);
+    auto resType = valC.getType();
+    Value zeroFlag = b.i32_val(0);
+    OperationState loweredOp(loc, mfmaInsn.getInsnName());
+    int32_t cbsz = getMfmaF8F6F4MatrixFormat(mfmaInsn.getElementTypeA());
+    int32_t blgp = getMfmaF8F6F4MatrixFormat(mfmaInsn.getElementTypeB());
+    assert((cbsz != -1) && (blgp != -1));
+    loweredOp.addTypes(resType);
+    loweredOp.addOperands({valA, valB, valC, b.i32_val(cbsz), b.i32_val(blgp),
+                           zeroFlag, valScaleA, zeroFlag, valScaleB});
+    return rewriter.create(loweredOp)->getResult(0);
+  }
+
+  LogicalResult convertScaledDot(DotScaledOp op,
+                                 DotScaledOpAdaptor adaptor) const {
+    // Check if this dot has come with priority set by setprio.
+    auto setPrioOp = dyn_cast_or_null<ROCDL::SetPrioOp>(op->getPrevNode());
+
+    auto warpsPerCTA = mfmaLayout.getWarpsPerCTA();
+    auto mDim = mfmaLayout.getMDim();
+    auto nDim = mfmaLayout.getNDim();
+    auto mfmaVersion = mfmaLayout.getVersionMajor();
+    assert((mDim == nDim && (mDim == 32 || mDim == 16 || mDim == 4)) ||
+           (mDim == 64 && nDim == 4) || (mDim == 4 && nDim == 64));
+
+    Value a = op.getLhs();
+    Value b = op.getRhs();
+    Value aScale = op.getLhsScale();
+    Value bScale = op.getRhsScale();
+    Value d = op.getD();
+    auto aTensorTy = cast<RankedTensorType>(a.getType());
+    auto bTensorTy = cast<RankedTensorType>(b.getType());
+    auto dTensorTy = cast<RankedTensorType>(d.getType());
+    auto elemTyA = aTensorTy.getElementType();
+    auto elemTyB = bTensorTy.getElementType();
+    ScaleDotElemType aElemType = op.getLhsType();
+    ScaleDotElemType bElemType = op.getRhsType();
+
+    auto supportsTypes = [](ScaleDotElemType elemType) {
+      return elemType == ScaleDotElemType::E2M1;
+    };
+
+    if (!supportsTypes(aElemType) || !supportsTypes(bElemType)) {
+      llvm::report_fatal_error("NYI: mxfp6, mxfp8\n");
+    }
```

Comment:
The functor seems unncessary. Also for error messages typically we want to be verbose and standalone. Image you are debugging somebody's codebase and see `NYI: mxfp, mxfp8` it would be not very straightforward. Something like `NYI: converting mxfp6/mxfp8 scaled dot to LLVM` would provide more context. :)

Reply:
Actually nvm. I think you will extend later with more element type cases..